src/main/java/com/treasure/hunt/strategy/hint/HalfplaneHint.java

```diff
@@ -0,0 +1,57 @@
+package com.treasure.hunt.strategy.hint;
+
+import lombok.Value;
+import org.locationtech.jts.algorithm.Angle;
+import org.locationtech.jts.geom.Point;
+
+@Value
+public class HalfplaneHint extends Hint {
+
+    public enum Direction{
+        right, left, up, down;
+    }
+
+    Point halfplanePointOne;
+    Point halfplanePointTwo;
+    Direction direction;    // when the line indicated by halfplanePointOne and halfplanePointTwo is not horizontal,
+                            // right and left indicate where the target is (right indicates the target is in positive x-Direction
+                            // in relationship to the line)
+                            // when the line is horizontal, up signals the target is in positive y-Direction in relationship
+                            // to the line (the up and down enumerators are only used when the line is horizontal)
+                            // left and down respectively
+
+    public static HalfplaneHint angular2correctHalfPlaneHint(AngleHint anglehint)
+    {
+
+
```

Comment:
Plase pull latest master, we have committed some code styling preferences to enhance readabilty. Also, search for shortcuts reformatting your code. More than 1 blank line inside of code declaration is disabled by default.